### PR TITLE
UX-2065, UX-2035 - Made the etcd backup and node upgrade startegy cha…

### PR DESCRIPTION
**Pull request by patil-pratik-87**
Screenshot 2022-08-24 at 3 15 45 PM
Screenshot 2022-08-24 at 3 20 08 PM
Screenshot 2022-08-24 at 3 20 16 PM
Screenshot 2022-08-24 at 3 20 45 PM
…nges for capi cluster create

### CODE REVIEW

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+const useStyles = makeStyles((theme: Theme) => ({
+  paramFields: {
+    display: 'grid',
+    gridGap: theme.spacing(2),
+  },
+}))
+
+const defaultEtcdBackup = [
+  {
+    name: 'daily',
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:21]: I think the name is capitalized according to Erwin

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    target: 'local',
+    localPath: '/etc/pf9/etcd-backup',
+    dailyBackupTime: '00:10',
+    backupCount: 5,
+  },
+  {
+    name: 'interval',
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:28]: I think the name is capitalized according to Erwin

File: src/app/utils/YamlTemplateParser.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export default class YamlTemplateParser {
 
   private parse(obj) {
     return pipe(
-      tap((value) => console.log(value)),
+      // tap((value) => console.log(value)), // uncomment to debug
```
Comment (knnguy) [inline src/app/utils/YamlTemplateParser.ts:21]: Do you still need this comment?

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  (minutes) => minutes > 9 && minutes < 60,
+  'Minutes must be between 10 and 59',
+)
+
+export default function CapiEtcdBackupFields({ wizardContext, setWizardContext }) {
+  const classes = useStyles()
+
+  const defaultDailyBackup = useMemo(() => {
+    const etcdBackup = wizardContext?.etcdBackup || defaultEtcdBackup
+    return etcdBackup.find((backup) => backup?.name === 'daily')
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:67]: Name is capitalized. Daily

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  }, [wizardContext?.etcdBackup])
+
+  const deafultIntervalBackup = useMemo(() => {
+    const etcdBackup = wizardContext?.etcdBackup || defaultEtcdBackup
+    return etcdBackup.find((backup) => backup?.name === 'interval')
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:72]: Name is capitalized. Interval

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+export default function CapiEtcdBackupFields({ wizardContext, setWizardContext }) {
+  const classes = useStyles()
+
+  const defaultDailyBackup = useMemo(() => {
+    const etcdBackup = wizardContext?.etcdBackup || defaultEtcdBackup
+    return etcdBackup.find((backup) => backup?.name === 'daily')
+  }, [wizardContext?.etcdBackup])
+
+  const deafultIntervalBackup = useMemo(() => {
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:70]: typo. defaultIntervalBackup

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+      )
+
+      setIntervalBackup((prev) =>
+        unit === 'hours' ? assoc('intervalHours', 1, prev) : assoc('intervalMins', 30, prev),
+      )
+    },
+    [intervalBackup, intervalBackupUnit],
+  )
+
+  console.log('ETCD backup ', wizardContext?.etcdbackup)
```
Comment (knnguy) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:120]: Console.logs

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx (hunk 6 above)
Comment (kevinylee1991) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:70]: nitpick, default spelled incorrectly

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx (hunk 7 above)
Comment (kevinylee1991) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:120]: console.logs

**Review by knnguy — APPROVED** [review]

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiMachineDeploymentNodeUpdateStrategies.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      </Row>
+      <Row minItemWidth="300">
+        <DropdownField
+          DropdownComponent={Dropdown}
+          id="maxSurgeType"
+          label="Max Surge - Type"
+          value={maxSurgeType}
+          items={rollingUpdateTypeOptions}
+          onChange={(value) => {
+            setMaxSurgeType(value),
```
Comment (kevinylee1991) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiMachineDeploymentNodeUpdateStrategies.tsx:46]: are this comma and indent on the next line needed?

File: src/app/utils/YamlTemplateParser.ts (hunk 3 above)
Comment (kevinylee1991) [inline src/app/utils/YamlTemplateParser.ts:21]: is this meant to be commented out still?

**Review by kevinylee1991 — APPROVED** [review]

File: src/app/utils/YamlTemplateParser.ts (hunk 3 above)
Comment (patil-pratik-87) [reply src/app/utils/YamlTemplateParser.ts:21]: Yeah, it's helpful to debug when there is a failure in YAML parsing. Let's keep it for now

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx (hunk 6 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:70]: ohh thanks

File: src/app/utils/YamlTemplateParser.ts (hunk 3 above)
Comment (patil-pratik-87) [reply src/app/utils/YamlTemplateParser.ts:21]: Its on demand, if the parsing fails we can uncomment and debug

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx (hunk 1 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/aws/capi/CapiEtcdBackupFields.tsx:21]: Cool, thanks saw it now in the channel, and will make the change.

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiMachineDeploymentNodeUpdateStrategies.tsx (hunk 8 above)
Comment (patil-pratik-87) [reply src/app/plugins/infrastructure/components/clusters/aws/capi/CapiMachineDeploymentNodeUpdateStrategies.tsx:46]: Not interntional, its the formatter/linter :)

**Pull request merged**